### DISCUSSION TITLE
v9.4 (2022/08)

### DISCUSSION THREAD

**Discussion by Mizux** [google/or-tools#3425] [Announces]
# Platforms
* Add Debain-10 support (#3209)
* Add Ubuntu 22.04 LTS support (#3276)
  * note: won't have .Net 3.1 support (see: https://github.com/dotnet/core/issues/7038#issuecomment-1110377345) 
* Remove Ubuntu 21.10 LTS support

# Misc
* Split archive by languages and add CMake config to the C++ one (#3200)

# Graph
Split ortools.graph.pywrapgraph into:
- ortools.graph.python.linear_sum_assignment
- ortools.graph.python.max_flow
- ortools.graph.python.min_cost_flow

This allows using numpy to speed up the setup of the problems.

# CP-SAT
Some improvement on:
* scheduling (propagation, cuts, lower bounds)
* MaxSAT (presolve, core based heuristics)
* MIP performance (presolve, cuts)

<hr /><em>This discussion was created from the release <a href='https://github.com/google/or-tools/releases/tag/v9.4'>v9.4 (2022/08)</a>.</em>

**Comment by jmarca** [comment]
Question about the release artifacts.  For previous releases (9.3, etc) the instructions for installing for C++ on linux from binary said to download files such as:  `...or-tools/releases/download/v9.3/or-tools_amd64_debian-11_v9.3.10497.tar.gz`

I don't see those kinds of files in the new 9.4 releases page.  Are they going to be generated in the near future, or is there a different way to install for C++ that I am not seeing.

**Comment by Mizux** [comment]
v9.4 upload status:
* pypi apple M1/x86-64 mismatch seems wheel package are named universal2 by newer Cython, on my way to fix it
EDIT: tracked by https://github.com/google/or-tools/issues/3436
* java: should be living on maven https://mvnrepository.com/artifact/com.google.ortools/ortools-java/9.4.1874
* .Net: seems I don't have the right to upload our two new runtimes (`linux-arm64` and `osx-arm64`) need to figure out why (we should have ownership on `Google.OrTools.*` :person_shrugging: )
* @jmarca other artifacts: will try to upload all I can ASAP
note: we now have one archive per language (except python) -> cpp, java, dotnet
note2:  contrary to maven/nuget artifact archive contains meta package which only depends on the target. eg `ortools_java_visualStudio2022.zip` will only contains and depends on `Google.OrTools.runtime-win32-x86_64` -> you won't pay 5x the runtime cost...
